File: CS471/fileConvers/controlBlock.py
#######################################################################################################################
#
# Andrew Baca
# October 29th, 2018
# controlBlock.py
# CS471
# 
# Purpose: this program removes unwanted data from the text file in the pattern of control-C to control-B
#	   and everything in between
#
######################################################################################################################
import sys						#import to write in terminal without newlines unless triggered
flag = 1;						#declare flag, set to 1
fw = open("update.txt", "w")				#open file to write to
with open("control-char.txt") as f:			#open file to read from
	while True:					#loop while input
		c = f.read(1)				#read in single char
		ascii = ord(c)				#convert char to ascii
		
		if not c:				#THROW eof ERROR MESSAGE
			print "End of file"	
			break

		if(ascii == 2 or ascii == 3):		#set flags according to controls
			if(ascii == 3):
				flag = 0
			if(ascii == 2):
				flag = 1

		if(flag == 1 and ascii != 2):		#write to file based on flag val
			sys.stdout.write(c)
			fw.write(c)



		

File: CS471/fileConvers/controlBlock.pl
################################################################################################################
#
# Andrew Baca
# controlBlock.pl
# October 29th, 2018
#
# CS 471
# Purpose: this program removes unwanted data from the text file in the pattern of control-C to control-B
#	   and everything in between
#
###############################################################################################################

open FILE1, "control-char.txt" or die $!;			#open or kill readable file
open(FH, '>', "update.txt") or die $!;				#open or kill writable file
my ($buf, $data, $n, $hold, $flag);				#declare and instantiate variables
$flag = 1;
while (($n = read FILE1, $data, 1) != 0)			#while loop to read through file
{
	print "$data read ";					#print char to terminal	
	$buf .= $data;						#print companion ascii value to terminal
	$hold = ord($data);
	print "$hold \n";
	if($hold == 2 || $hold == 3){				#set flags correspondingt to controls
		if($hold == 2){
			$flag = 1;
			#print "*************";
		}
		if($hold == 3){
			$flag = 0;
		}
	}
	if($flag == 1 && $hold != 2){				#if correct flag set, write to file
		print FH $data;
	}
	
}
close(FILE1);							#close files
close(FILE2);

